openedx/features/course_experience/static/course_experience/js/CourseGoals.js

```diff
@@ -0,0 +1,40 @@
+/* globals gettext */
+
+export class CourseGoals {  // eslint-disable-line import/prefer-default-export
+
+  constructor(options) {
+    $('.goal-option').click((e) => {
+      const goalKey = $(e.target).data().choice;
+      $.ajax({
+        method: 'POST',
+        url: options.setGoalUrl,
+        headers: { 'X-CSRFToken': $.cookie('csrftoken') },
```

Comment:
@nasthagiri Do you remember if we don't need this for DRF and why it is not needed with DRF?